addons/account_edi_ubl_cii/models/account_edi_common.py

```diff
@@ -0,0 +1,592 @@
+        Indeed, 30 $ / 3 pieces = 10 $ / piece => 10 * 3 (billed quantity) - 2 (rebate) = 28
+
+        UBL ROUNDING: "the result of Item line net
+            amount = ((Item net price (BT-146)÷Item price base quantity (BT-149))×(Invoiced Quantity (BT-129))
+        must be rounded to two decimals, and the allowance/charge amounts are also rounded separately."
+        It is not possible to do it in Odoo.
+
+        :params tree
+        :params xpath_dict dict: {
+            'basis_qty': list of str,
+            'gross_price_unit': str,
+            'rebate': str,
+            'net_price_unit': str,
+            'billed_qty': str,
+            'allowance_charge': str, to be used in a findall !,
+            'allowance_charge_indicator': str, relative xpath from allowance_charge,
+            'allowance_charge_amount': str, relative xpath from allowance_charge,
+            'line_total_amount': str,
+        }
+        :params: invoice_line_form
+        :params: qty_factor
+        """
+        # basis_qty (optional)
+        basis_qty = 1
+        for xpath in xpath_dict['basis_qty']:
+            basis_quantity_node = tree.find(xpath)
+            if basis_quantity_node is not None:
+                basis_qty = float(basis_quantity_node.text)
+
+        # gross_price_unit (optional)
+        gross_price_unit = None
+        gross_price_unit_node = tree.find(xpath_dict['gross_price_unit'])
+        if gross_price_unit_node is not None:
+            gross_price_unit = float(gross_price_unit_node.text)
+
+        # rebate (optional)
+        # Discount. /!\ as no percent discount can be set on a line, need to infer the percentage
+        # from the amount of the actual amount of the discount (the allowance charge)
+        rebate = 0
+        rebate_node = tree.find(xpath_dict['rebate'])
+        net_price_unit_node = tree.find(xpath_dict['net_price_unit'])
+        if rebate_node is not None:
+            if net_price_unit_node is not None and gross_price_unit_node is not None:
+                rebate = float(gross_price_unit_node.text) - float(net_price_unit_node.text)
+            else:
+                rebate = float(rebate_node.text)
+
+        # net_price_unit (mandatory)
+        net_price_unit = None
+        if net_price_unit_node is not None:
+            net_price_unit = float(net_price_unit_node.text)
+
+        # billed_qty (mandatory)
+        billed_qty = 1
+        product_uom_id = None
+        quantity_node = tree.find(xpath_dict['billed_qty'])
+        if quantity_node is not None:
+            billed_qty = float(quantity_node.text)
+            uom_xml = quantity_node.attrib.get('unitCode')
+            if uom_xml:
+                uom_infered_xmlid = [
+                    odoo_xmlid for odoo_xmlid, uom_unece in self._get_uom_unece_mapping().items() if uom_unece == uom_xml
+                ]
+                if uom_infered_xmlid:
+                    product_uom_id = self.env.ref('uom.' + uom_infered_xmlid[0])
```

Comment:
this is probably a bad idea. What if the uom are defined in another module? At least, use the `raise_if_not_found=False` keyword

Reply:
I've added the uom prefix to all xmlid in the global variable